### PR TITLE
Native `DataTransfer#files` should not be accessed when not needed or not relevant

### LINKED ISSUE #13366: Image drag & drop extremely sluggish in Firefox 109
## 📝 Provide detailed reproduction steps (if any)

1. Open the CKEditor 5 image demo page at https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html in Firefox 109.0.1
2. Click the image and drag it around the document

### ✔️ Expected result

The caret should move around the page quickly to show where the image will be dropped

### ❌ Actual result

The caret moves extremely slowly, lagging behind the movement of the cursor by several seconds

## 📃 Other details

This issue may be related to changes to `dataTransfer.files` in Firefox - https://bugzilla.mozilla.org/show_bug.cgi?id=1812611 

Here's the stack chart while dragging the image:

Screenshot 2023-01-31 at 4 46 52 pm

* Browser: Firefox 109.0.1
* OS: macOS 12.6.2

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (clipboard, engine): Dragging images in the editor should not lag in Firefox. Closes #13366.

---

### Additional information

### CODE REVIEW

File: packages/ckeditor5-engine/src/view/datatransfer.ts [hunk 1]
```diff
@@ -16,19 +16,34 @@ export default class DataTransfer {
...
+	private _files: Array<File> | null;
 
 	/**
 	 * The native DataTransfer object.
 	 */
 	private _native: DomDataTransfer;
 
-	constructor( nativeDataTransfer: DomDataTransfer ) {
-		this.files = getFiles( nativeDataTransfer );
+	constructor( nativeDataTransfer: DomDataTransfer, options: { cacheFiles?: boolean } = {} ) {
```
Comment (Reinmar) [inline packages/ckeditor5-engine/src/view/datatransfer.ts:26]: Doesn't the new option need to be documented?

Or it's blocked by the issue about porting API docs to TSDoc?

Comment (arkflpc) [reply packages/ckeditor5-engine/src/view/datatransfer.ts:26]: It's not blocked.

But, IMO, if the parameter name makes its purpose obvious, do we really need docs every time?

Comment (arkflpc) [reply packages/ckeditor5-engine/src/view/datatransfer.ts:26]: Second thought. It's not obvious, the cashing happens anyway. It should be called `eagerlyGetFiles` or `lazyGetFiles`, `cacheItNow` or something in this direction.

**Comment by niegowski** [comment]
@arkflpc can we merge this PR?

**Review by arkflpc — APPROVED** [review]

File: packages/ckeditor5-engine/src/view/datatransfer.ts (hunk 1 above)
Comment (Reinmar) [reply packages/ckeditor5-engine/src/view/datatransfer.ts:30]: I'd say that while the name says what it does. It does not say anything about why you may need that. And that's what the documentation should cover in most cases too.

**Pull request merged**